pkg/jsonpath/eval.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2024 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jsonpath
+
+// This file implements JSONPath querying almost as described in RFC 9535.
+// This file does not implement the function extensions.
+// The only comparisons implemented here are equality and inequality.
+// The order in which result nodes are produced is not consistent with the RFC.
+
+// JSONValue is something that can be produced by encoding/json.Unmarshal into a pointer
+// to a nil `any`.
+// That is: `bool`, `float64`, `string`, `nil`, `[]any`, or `map[string]any` --- where those
+// nested `any` have the same restriction.
+type JSONValue = any
+
+// Node is a JSON document node.
+// The other methods MUST NOT be invoked after Remove is invoked.
```

Comment:
You mean there is no  checking for the `nil`   value?  What happen if we do call it after Remove ?